syntax = "proto3";

enum ClientOperation {
    OP_CREATE_SESSION = 0;
    OP_JOIN_SPECIFIC_SESSION = 1;
    OP_JOIN_RANDOM_SESSION = 2;
}

message NewConnection {
    string name = 1;
    ClientOperation operation = 2; 
    int32 session_id = 3; // only used when operation is OP_JOIN_SPECIFIC_SESSION
}